lib/generateMarkdown.js

```diff
@@ -104,14 +108,39 @@ function createPropertiesSummary(schema, suppressWarnings) {
     return md;
 }
 
-function createPropertiesDetails(schema, title, headerLevel, suppressWarnings) {
-    var headerMd = style.getHeaderMarkdown(headerLevel);
+function createPatternPropertiesSummary(schema, suppressWarnings) {
     var md = '';
+    md += style.propertiesSummary('Pattern Properties') + '\n\n';
+    md += '|   |Type|Description|Required|\n';
+    md += '|---|----|-----------|--------|\n';
 
-    var properties = schema.properties;
+    var properties = schema.patternProperties;
     for (var name in properties) {
         if (properties.hasOwnProperty(name)) {
             var property = properties[name];
+            var summary = getPropertySummary(property, suppressWarnings);
+
+            md += '|' + style.propertyNameSummary(name) +
+                '|' + style.typeValue(summary.type) +
+                '|' + defaultValue(summary.description, '') +
+                '|' + (summary.required === 'Yes' ? style.requiredIcon : '') + summary.required + '|\n';
+        }
+    }
+    md += '\n';
+
+    return md;
+}
+
+var alreadyDetailed = []
+function createPropertiesDetails(schema, title, headerLevel, suppressWarnings, section) {
+    var headerMd = style.getHeaderMarkdown(headerLevel);
+    var md = '';
+    if (section === undefined) section = 'properties'
+    var properties = schema[section];
+    for (var name in properties) {
+        if (properties.hasOwnProperty(name) && alreadyDetailed.indexOf(name) === -1) {
+            alreadyDetailed.push(name);
```

Comment:
This isn't going to play nicely at all with the `typeLinking` PR that's out right now, since it generates the markdown for all related schemas.. this would exclude properties with the same name that other types have....